crates/orderbook/src/api/order_validation.rs

```diff
@@ -336,11 +362,35 @@ impl OrderValidating for OrderValidator {
             )
             .await
             .unwrap_or(false)
+        {
+            return Ok((order, unsubsidized_fee));
+        }
+
+        if !self
```

Comment:
Do you think it maybe makes sense to change `BalanceFetching` to have something like:
```rust
trait BalanceFetching {
    fn simulate_transfer(...) -> Result<(), TransferError>;
}
enum TransferError {
    InsufficientBalance,
    InsufficientAllowance,
    Other(...),
}
```

The advantage of doing it this way is that we keep the details of testing transfers and the possible places it could go wrong in the `BalanceFetching` component.

Some other advantages:
- It allows us to use a JSON RPC batch call for the balance and allowance query
- We won't accidentally forget to update `has_sufficient_balance` to also pass in the `SellTokenBalance` so that it works with Vault internal balances once we support it.